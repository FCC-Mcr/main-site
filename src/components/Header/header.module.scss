/**
 *  this is just for testing purposes to ensure scss and css modules
 *  are initialised correctly feel free to alter or remove this file.
 */

@import "../../styles/utility/responds-to.scss";
@import "../../styles/config/colour.scss";
@import "../../styles/config/globals.scss";

$header-height: 3.5rem;
$nav-spacing: 2rem;

.header {
  display: flex;
  align-items: center;
  background: rgba(255, 255, 255, 0.9);
  color: $accent;
  height: $header-height;
  justify-content: space-between;
  align-items: center;
  padding: 0 2em;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  @include responds-to("sm", "max") {
    padding: 0 1rem;
  }
  z-index: map-get($layers, "highest");

  > a {
    display: flex;
  }

  .logo {
    margin-bottom: 0;
    @include responds-to("sm", "max") {
      height: 2rem;
    }
  }
}

nav {
  a {
    color: $accent;
    font-weight: 600;
    margin-right: $nav-spacing;
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
  }
  :last-child {
    margin-right: 0;
  }
}

.nav {
  display: flex;
  align-items: center;
  @include responds-to("sm", "min") {
    img {
      display: none;
    }
  }
  @include responds-to("sm", "max") {
    > a {
      display: none;
    }
    img {
      height: 1.25rem;
      margin-bottom: 0;
    }
  }
}
