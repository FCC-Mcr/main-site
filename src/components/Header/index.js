import { Link } from "gatsby"
import PropTypes from "prop-types"
import React from "react"
import MobileMenu from "../MobileMenu"

// icons
import logo from "../../icons/logo.svg"
import styles from "./header.module.scss"

const activeStyles = {
  textDecoration: "underline",
}

const NavLink = ({ children, to }) => (
  <Link to={to} activeStyle={activeStyles}>
    {children}
  </Link>
)

const MainNav = () => (
  <nav className={styles.nav}>
    <NavLink to="/">Home</NavLink>
    <NavLink to="/blog/">Blog</NavLink>
    <NavLink to="/learning-resources/">Resources</NavLink>
    <NavLink to="/upcoming-meetups/">Upcoming Meetups</NavLink>
    <MobileMenu />
  </nav>
)

const Header = ({ siteTitle }) => (
  <>
    <header className={`${styles.header}  u-shadow--1`}>
      <Link to="/" className={styles.title} aria-label="Home">
        {/* need to bring in icon */}
        <img className={styles.logo} src={logo} alt="" />
      </Link>
      <MainNav />
    </header>
  </>
)

Header.propTypes = {
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
