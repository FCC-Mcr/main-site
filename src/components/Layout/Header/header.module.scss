/**
 *  this is just for testing purposes to ensure scss and css modules
 *  are initialised correctly feel free to alter or remove this file.
 */

@import "../../../styles/utility/responds-to.scss";
@import "../../../styles/config/colour.scss";
@import "../../../styles/config/globals.scss";

$header-height: 50px;
$nav-spacing: 30px;

.header {
  display: none;
  @include responds-to("xs") {
    display: flex;
  }
  align-items: center;
  background: $white;
  color: $accent;
  height: $header-height;
  justify-content: space-between;
  align-items: center;
  padding: 0 2em;
  position: relative;
  z-index: map-get($layers, "highest");

  .logo {
    margin-right: 10px;
    margin-bottom: -2px;
  }
}

nav {
  display: inline-block;
  font-size: 16px; // TODO: replace with configuration values

  a {
    color: $accent;
    height: 35px;
    font-weight: 700;
    margin-right: $nav-spacing;
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
  }
  :last-child {
    margin-right: 0;
  }
}
