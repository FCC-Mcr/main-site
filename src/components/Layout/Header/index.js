import { Link } from "gatsby"
import PropTypes from "prop-types"
import React from "react"

import FAB from "../../FAB"

// icons
import logo from "../../../icons/logo.svg"

import styles from "./header.module.scss"

const MainNav = () => (
  <nav>
    <a href="https://www.freecodecamp.org/forum/c/united-kingdom-study-groups/manchester">
      Forum
    </a>
    <Link to="/">Home</Link>
    <Link to="/blog/">Blog</Link>
    <Link to="/learning-resources/">Resources</Link>
    <Link to="/upcoming-talks/">Upcoming Talks</Link>
  </nav>
)

const Header = ({ siteTitle }) => (
  <>
    <header className={`${styles.header}  u-shadow--2`}>
      <h2>
        <Link to="/" className={styles.title}>
          {/* need to bring in icon */}
          <span className={styles.logo}>
            <img src={logo} alt="" />
          </span>
          {siteTitle}
        </Link>
      </h2>
      <MainNav />
    </header>
    <FAB />
  </>
)

Header.propTypes = {
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
