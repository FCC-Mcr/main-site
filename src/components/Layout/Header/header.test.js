import React from "react"
import renderer from "react-test-renderer"
import Header from "./index"
describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Header siteTitle="FCC MCR" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
