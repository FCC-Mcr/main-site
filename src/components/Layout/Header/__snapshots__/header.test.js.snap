// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Array [
  <header
    className="header  u-shadow--1"
  >
    <a
      aria-current="page"
      className="title"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      <img
        alt=""
        className="logo"
        src={Object {}}
      />
    </a>
    <nav>
      <a
        aria-current="page"
        className=""
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        href="/blog/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
      <a
        href="/learning-resources/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Resources
      </a>
      <a
        href="/upcoming-talks/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Upcoming Talks
      </a>
    </nav>
  </header>,
  <div
    className="fab u-shadow--3"
    onClick={[Function]}
  >
    <img
      alt="menu button"
      src={Object {}}
    />
  </div>,
]
`;
